File: webapi/Models/HR/Schedule.cs
using System.ComponentModel.DataAnnotations;

namespace webapi.Models.HR
{
    public class Schedule
    {
        [Key]
        public int Id { get; set; }
        public int EmployeeId { get; set; }
        public Employee? Employee { get; set; }
        public Int16 InsuranceDays { get; set; }
        public Int16 InsuranceExperienceDays { get; set; }
        public Int16 IncapacityDays { get; set; }
        public Int16 ChildcareDays { get; set; }
        public Int16 WithoutInsuranceDays { get; set; }
        public Int16 UnpaidLeaveDays { get; set; }
        public Int16 PaidIncapacityDays { get; set; }
        public Int16 WorkedHours { get; set; }
        public Int16 OvertimeHours { get; set; }
        public DateOnly CreationDate { get; set; }
    }

    public class ScheduleDTO
    {
        public int EmployeeId { get; set; }
        public Employee? Employee { get; set; }
        public Int16 InsuranceDays { get; set; }
        public Int16 InsuranceExperienceDays { get; set; }
        public Int16 IncapacityDays { get; set; }
        public Int16 ChildcareDays { get; set; }
        public Int16 WithoutInsuranceDays { get; set; }
        public Int16 UnpaidLeaveDays { get; set; }
        public Int16 PaidIncapacityDays { get; set; }
        public Int16 WorkedHours { get; set; }
        public Int16 OvertimeHours { get; set; }
    }
}

File: reactapp/src/models/HR/Schedule.ts
export interface Schedule {
    employeeId: int, 
    insuranceDays: int,
    insuranceExperienceDays: int,
    incapacityDays: int,
    childcareDays: int,
    withoutInsuranceDays: int,
    unpaidLeaveDays: int,
    paidIncapacityDays: int,
    workedHours: int,
    overtimeHours: int,
}


export const createDefaultSchedule = (employeeId: int): Schedule => {
    return {
        employeeId: employeeId,
        insuranceDays: 0,
        insuranceExperienceDays: 0,
        incapacityDays: 0,
        childcareDays: 0,
        withoutInsuranceDays: 0,
        unpaidLeaveDays: 0,
        paidIncapacityDays: 0,
        workedHours: 0,
        overtimeHours: 0,
    }
}